app/assets/javascripts/discourse/app/components/reviewable-user.gjs

```diff
@@ -1,89 +1,169 @@
 import Component from "@ember/component";
 import { concat } from "@ember/helper";
+import { on } from "@ember/modifier";
+import { action } from "@ember/object";
+import { service } from "@ember/service";
 import ReviewableField from "discourse/components/reviewable-field";
 import getUrl from "discourse/helpers/get-url";
-import discourseComputed from "discourse/lib/decorators";
+import rawDate from "discourse/helpers/raw-date";
+import { ajax } from "discourse/lib/ajax";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import discourseComputed, { bind } from "discourse/lib/decorators";
+import { REJECTED } from "discourse/models/reviewable";
 import { i18n } from "discourse-i18n";
+import ScrubRejectedUserModal from "admin/components/modal/scrub-rejected-user";
 
 export default class ReviewableUser extends Component {
+  @service currentUser;
+  @service modal;
+  @service store;
+
   @discourseComputed("reviewable.user_fields")
   userFields(fields) {
     return this.site.collectUserFields(fields);
   }
 
+  @discourseComputed("reviewable.status", "currentUser", "isScrubbed")
+  canScrubRejectedUser(status, currentUser, isScrubbed) {
+    return status === REJECTED && currentUser.admin && !isScrubbed;
+  }
+
+  @discourseComputed("reviewable.payload")
+  isScrubbed(payload) {
+    return !!payload?.scrubbed_by;
+  }
+
+  @action
+  showScrubRejectedUserModal() {
+    this.modal.show(ScrubRejectedUserModal, {
+      model: {
+        confirmScrub: this.scrubRejectedUser,
+      },
+    });
+  }
+
+  @bind
+  async scrubRejectedUser(reason) {
+    try {
+      await ajax({
+        url: `/review/${this.reviewable.id}/scrub`,
+        type: "PUT",
+        data: { reason },
+      });
+      this.store.find("reviewable", this.reviewable.id);
```

Comment:
I'm not familiar with the code here but is this done to refresh the reviewable in the store? I'm curious why we have to do this as we are not returning the reviewable object after finding it.

Reply:
Yah, it forces a refresh of the object.

I went with this approach after trying something like you suggested whilst I was working on #31901, but having trouble getting the object reliably updated with the correct data.